review/rolling.js

```diff
@@ -0,0 +1,58 @@
+import { CLASS_LIST } from '../util/data';
+import { li } from '../util/htmlTemplate';
+import { delay } from '../util/util.js';
+
+function RecommendRolling({ data, selector, animation }) {
+  this.rollingData = data;
+  this.rollingList = selector.rollingList;
+  this.searchInput = selector.searchInput;
+  this.step = 0;
+  this.oneStep = animation.oneStep;
+  this.transition = animation.transition;
+}
+
+RecommendRolling.prototype = {
+  constructor: RecommendRolling,
+  init() {
+    this.render();
+    this.autoRolling();
+  },
+  async autoRolling() {
+    if (this.step > this.rollingData.length) {
+      this.render();
+      this.step = 0;
+      await delay('', 0);
+      this.rolling();
+      this.autoRolling();
+    } else {
+      setTimeout(() => {
+        this.rolling();
+        this.autoRolling();
+      }, 3000);
+    }
+  },
+  rolling() {
+    this.step++;
+    this.setRollingAnimation({ moveY: this.oneStep * this.step * -1, transition: this.transition });
+  },
+  getrollingHTML() {
+    const { PLACEHOLDER_ITEM } = CLASS_LIST;
+    let rollingHTML = this.rollingData.reduce(
+      (acc, cur, idx) => acc + li({ value: `${idx + 1}. ${cur}`, classes: [PLACEHOLDER_ITEM] }),
+      ''
+    );
+    rollingHTML += li({ value: `1. ${this.rollingData[0]}`, classes: [PLACEHOLDER_ITEM] });
+    return rollingHTML;
```

Comment:
여기서는 reduce 함수를 잘 활용해 주셨네요 👍 지금도 충분히 괜찮은 코드이지만 조금 더 욕심을 내서 보조함수가 하는 일을 분리해서 간결하게 만들면 더욱 읽기 좋을 것 같아요.

```suggestion
const rollingHtmlList = this.rollingData
  .map((item, idx) => `${idx + 1}. ${item}`)
  .map(value => li({ value, classes: [PLACEHOLDER_ITEM] }));

return rollingHtmlList.join('') + rollingHtmlList[0];
```